I'm not sure why the build said it didn't find `node` but suspect it's not the core problem. Instead, the `nginx` installation failed because it couldn't find PCRE:
```
checking for PCRE library ... not found
checking for PCRE library in /usr/local/ ... not found
checking for PCRE library in /usr/include/pcre/ ... not found
checking for PCRE library in /usr/pkg/ ... not found
checking for PCRE library in /opt/local/ ... not found

./configure: error: the HTTP rewrite module requires the PCRE library.
You can either disable the module by using --without-http_rewrite_module
option, or install the PCRE library into the system, or build the PCRE library
statically from the source with nginx by using --with-pcre=<path> option.

make: *** No rule to make target 'build', needed by 'default'.  Stop.
make: *** No rule to make target 'install'.  Stop.
```

@BrennanConroy @anurse do we actually need the HTTP rewrite model? Otherwise, what package should we ask for? https://packages.ubuntu.com/xenial/pcre2-utils looks close but more low level than the agents usually deal with.

Of course, continuing after the error implies eng/scripts/install-nginx-linux.sh and probably install-nginx-mac.sh need `set -euo pipefail`. (Does this command work on macOS?)

--- a/eng/scripts/install-nginx-linux.sh
+++ b/eng/scripts/install-nginx-linux.sh
@@ -1,7 +1,9 @@
 #!/usr/bin/env bash
 
+set -euo pipefail
+
 curl -sSL http://nginx.org/download/nginx-1.14.2.tar.gz | tar zxfv - -C /tmp && cd /tmp/nginx-1.14.2/
-./configure --prefix=$HOME/nginxinstall --with-http_ssl_module
+./configure --prefix=$HOME/nginxinstall --with-http_ssl_module --without-http_rewrite_module
 make
 make install
 echo "##vso[task.prependpath]$HOME/nginxinstall/sbin"

